lib/main/atom/utils/atom.ts

```diff
@@ -97,3 +97,32 @@ async function editorTokenized(editor: Atom.TextEditor) {
     }
   })
 }
+
+// From https://github.com/atom-community/atom-ide-outline/blob/ec1a7197d63055de910562da3cc2b95fd939afc4/src/main.ts#L53
+// minimum number of line length to trigger large file optimizations
+const longLineLength =
+  (atom.config.get("linter-ui-default.longLineLength") as number | null) ?? 4000
```

Comment:
Grabbing configuration like this from other packages can be _very_ brittle. _Especially_ on the module level. I really don't like this, pretty sure it works as intended only whenever it feels like it.

A more robust approach that doesn't require drastic architectural changes is to defer initialization until the values are actually needed.

Also could be nice to update these via `atom.config.observe`

Also not a fan that this pulls config from `linter-ui-default` which isn't necessarily installed (I still kinda support atom-ide). Perhaps we can at expose these as configuration options in atom-typescript?

Reply:
I used linter-ui-default config because I want to eventually make this a central config. If we start adding this to every single package, it is really inconvenient for the users. If you want I can open a PR to add this config to atom-ide-base or atom itself.

Reply:
I understand the intention. However, this feels very arcane for the end-user. Here's a solution that will kind of work for now: expose these options in atom-typescript; set default to `0`, in the description explain that `0` pulls from linter-ui-default or uses `X` value if that fails.

The best option is to have this exposed as core Atom settings (well, editor atom settings more precisely), but I don't know if we want to hold this off until it gets into a public Atom release even _if_ the pull request gets merged.

In principle, I don't mind pulling from `linter-ui-default` _temporarily_, but I fear this will end up haunting the code base for the years to come, so I'm a bit reluctant.

That all said, I'm more concerned about this code being brittle.